cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -1778,6 +1784,711 @@ class SortIndicesMetaFunction : public MetaFunction {
   }
 };
 
+// ----------------------------------------------------------------------
+// TopK/BottomK implementations
+
+using SelectKOptionsState = internal::OptionsWrapper<SelectKOptions>;
+const auto kDefaultTopKOptions = SelectKOptions::TopKDefault();
+const auto kDefaultBottomKOptions = SelectKOptions::BottomKDefault();
+
+const FunctionDoc top_k_doc(
+    "Return the indices that would partition an array array, record batch or table\n"
+    "around a pivot",
+    ("@TODO"), {"input", "k"}, "PartitionNthOptions");
```

Comment:
I realize these are TODOs but don't forget to update these :)

Reply:
Is `k` really a parameter? Shouldn't these be unary functions?

Reply:
I though that kernel arity contains the number of required arguments for the function. This function requires at least it is required the input and k.  So arity should be binary.

Reply:
But `k` is also in the options, so it's being passed twice.